### DISCUSSION TITLE
Don't apply future transactions until the date arrives?

### DISCUSSION THREAD

**Discussion by MattFaz** [firefly-iii/firefly-iii#5230] [Firefly III] [answered]
Hey!
I am moving over from YNAB, so far so good however I do have a question around future transactions.

As an example I have to pay Rates (tax) on my house quarterly, I get the dates and amounts at the start of the year and I add them as future transactions. So I could add $500 as a withdrawal dated for 30 November with the source account as my Credit Card.

In YNAB, my Credit Card would show $0 until the date is 30 Nov then it will show -$500, however firefly appears to show my Credit Card on -$500 immediately when it's the same month. Is there a way to make it not do this until the transaction date has arrived?

Thank you :)

**Comment by JC5** [comment, marked as answer]
This depends on the visible date range, which is probably 1-30 nov. In such cases, Firefly III will calculate the balance for the end of the month which includes "future" transactions.

If you look at the account at page `/accounts/show/xx`, you'll see the graph drop at the right moment.

**Reply by MattFaz** [reply]
Thanks JC! Good to know :)
I think a good alternative could be to make a default date range that is month to date.
Create a feature request here: https://github.com/firefly-iii/firefly-iii/issues/5234

**Comment by MattFaz** [comment]
Created a feature reqeust that could resolve this discussion: https://github.com/firefly-iii/firefly-iii/issues/5234
